### DISCUSSION TITLE
如何在serverless中获取当前的consumer信息

### DISCUSSION THREAD

**Discussion by heichong** [apache/apisix#10116] [Q&A] [answered]
目的：
自定义serverless函数，根据不同的消费者，向upstream传递不同的参数。

问题：
1. 如何在serverless获取当前consumer的信息（如username）
2. 我想在consumer的配置中，增加自定义的一些属性，比如为每个consumer都增加testName,TestId，在serverless中把当前consumer的testName,TestId属性传递到upstream。不知道这种方式是否支持？

感谢！

**Comment by kayx23** [comment, marked as answer]
See if you can get `consumer_name` in the serverless function: https://docs.api7.ai/apisix/reference/built-in-variables#apisix-variables

**Reply by heichong** [reply]
非常感谢

如果我在创建consumer的配置上增加 : `testName` ，大概长这样：
```
{
  "username": "epc",
  "desc": "建管平台",
  "testName":"hello",
  "plugins": {
    "key-auth": {
      "_meta": {
        "disable": false
      },
      "key": "xxxxx"
    }
  }
}
```
在serverless中如何获取到这个testName属性？

**Reply by kayx23** [reply]
There's a serverless example here that returns `route.labels` where `labels` is a custom field on a route: https://docs.api7.ai/apisix/reference/built-in-variables#method-2-in-the-serverless-plugins

I believe you could tweak to your use case :)

**Reply by heichong** [reply]
好的，多谢回复
